### PR TITLE
Added ClientID to Ticket to prompt redirect after password change

### LINKED ISSUE #253: Ticket missing field ClientId for ChangePassword and VerifyEmail
### Description

According to the Auth0 [documentation](https://auth0.com/docs/api/management/v2#!/Tickets/post_password_change), the endpoint to request a password change and verify email should be able to take an additional argument of ClientID

client_id: ID of the client. If provided for tenants using New Universal Login experience, the user will be prompted to redirect to the default login route of the corresponding application once the ticket is used. See Configuring Default Login Routes for more details.

Currently, this library does not support passing in this argument, and as such removes the option for being navigated to the correct login page.

### Affected Resources

* `management.Ticket`

### References

Documentation of Ticket endpoints:
https://auth0.com/docs/api/management/v2/#!/Tickets/post_email_verification
https://auth0.com/docs/api/management/v2/#!/Tickets/post_password_change

Similar GitHub issues within other language SDKs:
https://github.com/auth0/auth0-java/issues/351
https://github.com/auth0/auth0.net/issues/463 

* #0000

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by 1pavanb**
### Proposed Changes

* Added ClientID to Ticket to prompt redirect after password change
* Fixed typos in comments

Fixes #253 

#### Acceptance Test Output

<details><summary>$ go test ./... -v -run TestUser</summary>

```
testing: warning: no tests to run
PASS
ok      gopkg.in/auth0.v5       (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      gopkg.in/auth0.v5/internal/client       (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      gopkg.in/auth0.v5/internal/tag  (cached) [no tests to run]
testing: warning: no tests to run
PASS
ok      gopkg.in/auth0.v5/internal/testing/expect       (cached) [no tests to run]
=== RUN   TestUser
=== RUN   TestUser/MarshalJSON
=== RUN   TestUser/UnmarshalJSON
=== RUN   TestUser/Create
    user_test.go:119: {
          "user_id": "auth0|61d6c5044bf551006fc805dd",
          "connection": "Username-Password-Authentication",
          "email": "chuck@chucknorris.com",
          "name": "chuck@chucknorris.com",
          "given_name": "Chuck",
          "family_name": "Norris",
          "username": "chucknorris",
          "nickname": "Chucky",
          "password": "Passwords hide their Chuck",
          "created_at": "2022-01-06T10:31:32.262Z",
          "updated_at": "2022-01-06T10:31:32.262Z",
          "user_metadata": {
            "favourite_attack": "roundhouse_kick"
          },
          "identities": [
            {
              "connection": "Username-Password-Authentication",
              "provider": "auth0",
              "isSocial": false,
              "user_id": "61d6c5044bf551006fc805dd"
            }
          ],
          "verify_email": false,
          "app_metadata": {
            "facts": [
              "count_to_infinity_twice",
              "kill_two_stones_with_one_bird",
              "can_hear_sign_language",
              "knows_victorias_secret"
            ]
          },
          "picture": "https://example-picture-url.jpg",
          "blocked": false,
          "email_verified": true
        }
=== RUN   TestUser/Read
    user_test.go:127: {
          "user_id": "auth0|61d6c5044bf551006fc805dd",
          "email": "chuck@chucknorris.com",
          "name": "chuck@chucknorris.com",
          "given_name": "Chuck",
          "family_name": "Norris",
          "username": "chucknorris",
          "nickname": "Chucky",
          "created_at": "2022-01-06T10:31:32.262Z",
          "updated_at": "2022-01-06T10:31:32.262Z",
          "user_metadata": {
            "favourite_attack": "roundhouse_kick"
          },
          "identities": [
            {
              "connection": "Username-Password-Authentication",
              "provider": "auth0",
              "isSocial": false,
              "user_id": "61d6c5044bf551006fc805dd"
            }
          ],
          "app_metadata": {
            "facts": [
              "count_to_infinity_twice",
              "kill_two_stones_with_one_bird",
              "can_hear_sign_language",
              "knows_victorias_secret"
            ]
          },
          "picture": "https://example-picture-url.jpg",
          "blocked": false,
          "email_verified": true
        }
=== RUN   TestUser/List
    user_test.go:135: {
          "start": 0,
          "limit": 50,
          "length": 2,
          "total": 2,
          "users": [
            {
              "user_id": "auth0|61d6c5044bf551006fc805dd",
              "email": "chuck@chucknorris.com",
              "name": "chuck@chucknorris.com",
              "given_name": "Chuck",
              "family_name": "Norris",
              "username": "chucknorris",
              "nickname": "Chucky",
              "created_at": "2022-01-06T10:31:32.262Z",
              "updated_at": "2022-01-06T10:31:32.262Z",
              "user_metadata": {
                "favourite_attack": "roundhouse_kick"
              },
              "identities": [
                {
                  "connection": "Username-Password-Authentication",
                  "provider": "auth0",
                  "isSocial": false,
                  "user_id": "61d6c5044bf551006fc805dd"
                }
              ],
              "app_metadata": {
                "facts": [
                  "count_to_infinity_twice",
                  "kill_two_stones_with_one_bird",
                  "can_hear_sign_language",
                  "knows_victorias_secret"
                ]
              },
              "picture": "https://example-picture-url.jpg",
              "blocked": false,
              "email_verified": true
            }
          ]
        }
=== RUN   TestUser/Update
    user_test.go:147: {
          "user_id": "auth0|61d6c5044bf551006fc805dd",
          "connection": "Username-Password-Authentication",
          "email": "chuck@chucknorris.com",
          "name": "chuck@chucknorris.com",
          "given_name": "Chuck",
          "family_name": "Norris",
          "username": "chucknorris",
          "nickname": "Chucky",
          "password": "I don't need one",
          "created_at": "2022-01-06T10:31:32.262Z",
          "updated_at": "2022-01-06T10:31:32.262Z",
          "user_metadata": {
            "favourite_attack": "roundhouse_kick"
          },
          "identities": [
            {
              "connection": "Username-Password-Authentication",
              "provider": "auth0",
              "isSocial": false,
              "user_id": "61d6c5044bf551006fc805dd"
            }
          ],
          "app_metadata": {
            "facts": [
              "count_to_infinity_twice",
              "kill_two_stones_with_one_bird",
              "can_hear_sign_language",
              "knows_victorias_secret"
            ]
          },
          "picture": "https://example-picture-url.jpg",
          "blocked": false,
          "email_verified": true
        }
=== RUN   TestUser/Update/AppMetadata
    user_test.go:160: {
          "user_id": "auth0|61d6c5044bf551006fc805dd",
          "connection": "Username-Password-Authentication",
          "email": "chuck@chucknorris.com",
          "name": "chuck@chucknorris.com",
          "given_name": "Chuck",
          "family_name": "Norris",
          "username": "chucknorris",
          "nickname": "Chucky",
          "created_at": "2022-01-06T10:31:32.262Z",
          "updated_at": "2022-01-06T10:31:33.481Z",
          "user_metadata": {
            "favourite_attack": "roundhouse_kick"
          },
          "identities": [
            {
              "connection": "Username-Password-Authentication",
              "provider": "auth0",
              "isSocial": false,
              "user_id": "61d6c5044bf551006fc805dd"
            }
          ],
          "app_metadata": {
            "facts": [
              "count_to_infinity_twice",
              "kill_two_stones_with_one_bird",
              "can_hear_sign_language",
              "knows_victorias_secret"
            ],
            "foo": "bar"
          },
          "picture": "https://example-picture-url.jpg",
          "blocked": false,
          "email_verified": true
        }
=== RUN   TestUser/Roles
    user_test.go:169: []
=== RUN   TestUser/AssignRoles
=== RUN   TestUser/RemoveRoles
=== RUN   TestUser/Permissions
    user_test.go:193: []
=== RUN   TestUser/AssignPermissions
=== RUN   TestUser/RemovePermissions
=== RUN   TestUser/Blocks
    user_test.go:223: []
=== RUN   TestUser/BlocksByIdentifier
    user_test.go:231: []
=== RUN   TestUser/Unblock
=== RUN   TestUser/UnblockByIdentifier
=== RUN   TestUser/Enrollments
    user_test.go:253: []
=== RUN   TestUser/RegenerateRecoveryCode
    user_test.go:261: {
          "recovery_code": "X43TGZVHPQP3N8U8NKJ3KE3G"
        }
=== RUN   TestUser/InvalidateRememberBrowser
=== RUN   TestUser/Delete
=== RUN   TestUser/Search
    user_test.go:319: {
          "start": 0,
          "limit": 50,
          "length": 1,
          "total": 1,
          "users": [
            {
              "user_id": "auth0|61d6c50a4bf551006fc805e0",
              "email": "alice@example.com",
              "name": "alice@example.com",
              "username": "alice",
              "nickname": "alice",
              "created_at": "2022-01-06T10:31:38.502Z",
              "updated_at": "2022-01-06T10:31:38.502Z",
              "identities": [
                {
                  "connection": "Username-Password-Authentication",
                  "provider": "auth0",
                  "isSocial": false,
                  "user_id": "61d6c50a4bf551006fc805e0"
                }
              ],
              "picture": "https://s.gravatar.com/avatar/c160f8cc69a4f0bf2b0362752353d060?s=480\u0026r=pg\u0026d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fal.png",
              "email_verified": false
            }
          ]
        }
=== RUN   TestUser/ListByEmail
    user_test.go:330: [{
          "user_id": "auth0|61d6c50a4bf551006fc805e0",
          "email": "alice@example.com",
          "name": "alice@example.com",
          "username": "alice",
          "nickname": "alice",
          "created_at": "2022-01-06T10:31:38.502Z",
          "updated_at": "2022-01-06T10:31:38.502Z",
          "identities": [
            {
              "connection": "Username-Password-Authentication",
              "provider": "auth0",
              "isSocial": false,
              "user_id": "61d6c50a4bf551006fc805e0"
            }
          ],
          "picture": "https://s.gravatar.com/avatar/c160f8cc69a4f0bf2b0362752353d060?s=480\u0026r=pg\u0026d=https%3A%2F%2Fcdn.auth0.com%2Favatars%2Fal.png",
          "email_verified": false
        }]
=== RUN   TestUser/Link
    user_test.go:367: [{0xc00020a830 0xc00020a860 0xc00020a850 0xc0000bcb8d <nil> <nil> <nil>} {0xc00020a880 0xc00020a8b0 0xc00020a8a0 0xc0000bcbcd <nil> <nil> <nil>}]
--- PASS: TestUser (21.73s)
    --- PASS: TestUser/MarshalJSON (0.00s)
    --- PASS: TestUser/UnmarshalJSON (0.00s)
    --- PASS: TestUser/Create (0.46s)
    --- PASS: TestUser/Read (0.26s)
    --- PASS: TestUser/List (0.28s)
    --- PASS: TestUser/Update (0.68s)
        --- PASS: TestUser/Update/AppMetadata (0.27s)
    --- PASS: TestUser/Roles (0.26s)
    --- PASS: TestUser/AssignRoles (0.28s)
    --- PASS: TestUser/RemoveRoles (0.24s)
    --- PASS: TestUser/Permissions (0.24s)
    --- PASS: TestUser/AssignPermissions (0.54s)
    --- PASS: TestUser/RemovePermissions (0.25s)
    --- PASS: TestUser/Blocks (0.29s)
    --- PASS: TestUser/BlocksByIdentifier (0.42s)
    --- PASS: TestUser/Unblock (0.63s)
    --- PASS: TestUser/UnblockByIdentifier (0.25s)
    --- PASS: TestUser/Enrollments (0.29s)
    --- PASS: TestUser/RegenerateRecoveryCode (0.34s)
    --- PASS: TestUser/InvalidateRememberBrowser (0.26s)
    --- PASS: TestUser/Delete (0.33s)
    --- PASS: TestUser/Search (0.31s)
    --- PASS: TestUser/ListByEmail (0.25s)
    --- PASS: TestUser/Link (9.51s)
=== RUN   TestUserIdentity
=== RUN   TestUserIdentity/MarshalJSON
=== RUN   TestUserIdentity/UnmarshalJSON
--- PASS: TestUserIdentity (0.00s)
    --- PASS: TestUserIdentity/MarshalJSON (0.00s)
    --- PASS: TestUserIdentity/UnmarshalJSON (0.00s)
PASS
ok      gopkg.in/auth0.v5/management    21.736s
```

</details>

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by 1pavanb** [comment]
is the accessor generator done by github workflow automatically?

File: management/ticket.go [hunk 1]
```diff
@@ -13,6 +13,13 @@ type Ticket struct {
 	// applied.
 	TTLSec *int `json:"ttl_sec,omitempty"`
 
+	// ID of the client. If provided for tenants using New Universal Login experience,
+	// the user will be prompted to redirect to the default login route of the
+	// corresponding application once the ticket is used.
+	//
+	// Conflicts with: ResultURL
+	ClientID string `json:"client_id,omitempty"`
```
Comment (apamildner) [inline management/ticket.go:21]: Shouldn't this be `*string` to reflect that it is optional? Right now it will break the API if I'm not mistaken.

Comment (1pavanb) [reply management/ticket.go]: yes, thank you

**Comment by marilangaas** [comment]
Any updates on this? We would really like this feature @alexkappa

**Comment by 1pavanb** [comment]
@marilangaas 
not ideal but there's a workaround
https://community.auth0.com/t/password-change-ticket-missing-clientid/48439/4

**Comment by 1pavanb** [comment]
> Any updates on this? We would really like this feature @alexkappa

@marilangaas 
not ideal but there's a workaround
https://community.auth0.com/t/password-change-ticket-missing-clientid/48439/4

**Comment by sergiught** [comment]
Hi @1pavanb, thanks for your contribution. This project is now being maintained at https://github.com/auth0/go-auth0 and unfortunately we are unable to transfer open PRs. Would you mind resubmitting this against that repository? 🙏🏻

**Pull request closed without merging**